docs/fsharp-cheatsheet.md

```diff
@@ -1151,11 +1170,10 @@ module Groceries =
 let fruit = Banana
 ```
 
-*However*, `AutoOpen` should be used cautiously. When an `open` or `AutoOpen` is used, all declarations in the containing element
-will be brought into scope. This can lead to [shadowing](https://en.wikipedia.org/wiki/Variable_shadowing); where the last
-named declaration replaces all prior identically-named declarations. There is *no* error - or even a warning - in F#, when shadowing occurs.
-A [coding convention (MS Learn)](https://learn.microsoft.com/en-us/dotnet/fsharp/style-guide/conventions#sort-open-statements-topologically) exists for `open`
-statements to avoid pitfalls; `AutoOpen` would sidestep this.
+#### Use `AutoOpen` and/or ignore `RequireQualifiedAccess` best practice with caution!
+
+Functions with identical names from different modules will silently "shadow" each other, causing the most recently imported definition to be used instead of the one you might expect. **The compiler will _not_ warn you that this has happened.** Even if your code works, you may encounter compiler errors if the `open` order changes. A [coding convention (MS Learn)](https://learn.microsoft.com/en-us/dotnet/fsharp/style-guide/conventions#sort-open-statements-topologically) exists for `open` statements to avoid pitfalls.
```

Comment:
this warning about silent meaning changes is not inaccurate, but I can imagine it scaring people into thinking they might as well be using something typeless. I dont know how I'd convey it, but the only real problem it triggers is inferring the wrong type where it's not pinned, which then manifests as an error some distance away in terms of files or compile time.